### PR TITLE
Adds release workflow for windows and tarball

### PR DESCRIPTION
**Pull request by gaiksaya**
Wont be able to test it as it is release (prod) related

### CODE REVIEW

File: .github/workflows/release-sync-tarball.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Sync tar artifacts to Prod
+on: 
+  repository_dispatch:
+    types: [sync-tar-prod]
+jobs:
+  Sync-Deb-Prod:
```
Comment (peterzhuamazon) [inline .github/workflows/release-sync-tarball.yml:6]: Name error

File: .github/workflows/release-sync-windows.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: Sync windows artifacts to Prod
+on: 
+  repository_dispatch:
+    types: [sync-windows-prod]
+jobs:
+  Sync-Deb-Prod:
```
Comment (peterzhuamazon) [inline .github/workflows/release-sync-windows.yml:6]: Name error

Comment (gaiksaya) [reply .github/workflows/release-sync-windows.yml:6]: > Name error

Corrected

File: .github/workflows/release-sync-tarball.yml (hunk 1 above)
Comment (gaiksaya) [reply .github/workflows/release-sync-tarball.yml:6]: > Name error

Corrected!

**Pull request merged**